src/Razor/src/Microsoft.AspNetCore.Razor.ProjectEngineHost/Utilities/StreamExtensions.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.ProjectSystem;
+
+#if !NET
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+#endif
+
+namespace Microsoft.AspNetCore.Razor.Utilities;
+
+internal static class StreamExtensions
+{
+    public static Task WriteStringAsync(this Stream stream, string text, Encoding? encoding = null, CancellationToken cancellationToken = default)
```

Comment:
Sorry I didn't notice this earlier, but I would very much prefer not optional parameters on internal methods.